File: lab/python/textos/contagem_de_carecteres.py
def contagem1(string):
    return len(string)


def contagem2(string):
    tam = 0
    for _ in string:
        tam += 1

    return tam


def contagem3(string):
    string += "\0"

    cont = 0
    while string[cont] != "\0":
        cont += 1
    return cont


string = input()
print(contagem1(string))
print(contagem2(string))
print(contagem3(string))

File: lab/c/textos/contagem_de_carecteres.c
#include <stdio.h>

// Versão 1: usando strlen (biblioteca padrão)
#include <string.h>
int contagem1(const char *string) {
    return strlen(string);
}

// Versão 2: contando manualmente
int contagem2(const char *string) {
    int tam = 0;
    while (string[tam] != '\0') {
        tam++;
    }
    return tam;
}

// Versão 3: loop parecido com a 2 (simulação manual)
int contagem3(const char *string) {
    int cont = 0;
    while (string[cont] != '\0') {
        cont++;
    }
    return cont;
}

int main() {
    char string[100];
    scanf("%99s", string); // lê string sem espaços

    printf("%d\n", contagem1(string));
    printf("%d\n", contagem2(string));
    printf("%d\n", contagem3(string));

    return 0;
}
